src/compiler/tree.cc

```diff
@@ -318,11 +380,40 @@ void TreeGrower::grow(Program* program) {
           method_queue.push_back(probe);
         }
       }
+      // Filtered selectors hit, if there is a class that has a method for it, and
+      // that method is in the filtered set.
+      filtered_selectors.for_each([&](CallSelector selector, UnorderedSet<Method*>& methods) {
+        if (methods.empty()) return;
+        auto probe = queryable.lookup(selector);
+        if (probe != null && methods.contains(probe)) {
+          logger->add_method_with_selector(selector, probe);
+          method_queue.push_back(probe);
+          methods.erase(probe);
+        }
+      });
     }
 
     found_selectors.erase_all(handled_selectors);
     handled_selectors.insert(found_selectors.begin(), found_selectors.end());
-    if (!found_selectors.empty()) {
+    bool has_filtered_selectors = false;
+    found_filtered_selectors.for_each([&](CallSelector selector, UnorderedSet<Method*>& methods) {
+      if (handled_selectors.contains(selector)) {
+        // All methods that fit this selector are handled anyway.
+        // No need to filter them.
```

Comment:
Should the `methods` set be erased -- or perhaps better yet should the entry for the selector in the `found_filtered_selectors` be cleared? It seems annoying that you have to do this here when you already did the thinning out earlier, but I guess the `handled_selectors` set may have grown.

Reply:
changed.